tests/e2e/targetallocator-prometheuscr/00-assert.yaml

```diff
@@ -27,24 +27,23 @@ metadata:
 data:
   collector.yaml: |
     exporters:
-      prometheus:
-        endpoint: 0.0.0.0:9090
-    processors: null
+        prometheus:
+            endpoint: 0.0.0.0:9090
     receivers:
-      prometheus:
-        config: {}
-        target_allocator:
-          collector_id: ${POD_NAME}
-          endpoint: http://prometheus-cr-targetallocator:80
-          interval: 30s
+        prometheus:
```

Comment:
is there a reason we need to change the whitespacing for these now? I saw the indent call above in the serialization and wasn't sure why we need that.

Reply:
y, thats a bit weird. [This one](https://github.com/open-telemetry/opentelemetry-operator/pull/2532/files#diff-395c66bb78902810e7c0ed75332a8b8dd0a404cf09306d0b9aa0e633ebfc3969R94) works fine. But as soon as we use `yaml.Marshal` it falls back to ident = 4.

Reply:
The default `yaml.Marshal` creates a `newEncoder()` which is initliazed with ident = 4.

https://github.com/go-yaml/yaml/blob/f6f7691b1fdeb513f56608cd2c32c51f8194bf51/encode.go#L56-L67